pkg/controller/jobframework/setup.go

```diff
@@ -99,6 +100,50 @@ func (m *integrationManager) setupControllers(mgr ctrl.Manager, log logr.Logger,
 	})
 }
 
+func (m *integrationManager) setupControllerAndWebhook(mgr ctrl.Manager, name string, fwkNamePrefix string, cb IntegrationCallbacks, options Options, opts ...Option) error {
+	if err := cb.NewReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, options.ManagerName)),
+		opts...,
+	).SetupWithManager(mgr); err != nil {
+		return fmt.Errorf("%s: %w", fwkNamePrefix, err)
+	}
+	if err := cb.SetupWebhook(mgr, opts...); err != nil {
+		return fmt.Errorf("%s: unable to create webhook: %w", fwkNamePrefix, err)
+	}
+	m.enableIntegration(name)
+	return nil
+}
+
+func waitForAPI(ctx context.Context, mgr ctrl.Manager, log logr.Logger, gvk schema.GroupVersionKind, retryInterval time.Duration, action func()) {
+	for {
+		_, err := getRESTMapping(mgr, gvk)
+		if err == nil {
+			action()
+			return
+		}
+		if !meta.IsNoMatchError(err) {
+			log.Error(err, "Failed to get REST mapping for gvk", "gvk", gvk)
+		}
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(retryInterval):
```

Comment:
Nit in terms of optimisation (can be done in follow up, not a blocker at all) - Given we are re-trying fetching the rest mapper in every cycle, better to use golang's [ticker](https://gobyexample.com/tickers):

```
ticker := time.NewTicker(retryInterval)
defer ticker.Stop()
```

Reason for this being - in this case a new timer is created every iteration, and it remains till the garbage collector does its job.

Reply:
That's an excellent idea, thanks!